core/commands/pin.go

```diff
@@ -332,6 +333,27 @@ Example:
 	},
 }
 
+var verifyPinCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Verify that recursive pins are complete.",
+	},
+	Run: func(req cmds.Request, res cmds.Response) {
+		n, _ := req.InvocContext().GetNode()
+
+		rdr, wtr := io.Pipe()
+		out := pinVerify(req.Context(), n)
+
+		go func() {
+			defer wtr.Close()
+			for r := range out {
+				r.Format(wtr)
+			}
+		}()
+
+		res.SetOutput(rdr)
```

Comment:
@whyrusleeping this is a WIP, more concerned with the U.I.  will fix up this later.